File: laravel/app/Http/Controllers/postController.php
<?php

namespace App\Http\Controllers;

use App\Models\Comment;
use App\Models\Post;
use App\Models\User;
use Illuminate\Http\Request;
class postController extends Controller
{
    //
  public function index(){


   $post=Post::with('comments','likes')->orderBy("created_at","desc")->get();
   return response()->json([
    "data"=> $post
   ]);

  }

  public function show($id){

    $post=Post::where('id',$id)->withCount('comments','likes')->with('comments','likes')->get();

    return response()->json([
        "post"=>$post,
    ]);
  }

  public function store(Request $request){

   $request->validate([
    "title"=>"required",
    "description"=>"required|string"
   ]);

   $post=Post::create([
    'title'=> $request->title,
    'description'=> $request->description,
    'user_id'=>auth()->user()->id

   ]);
   $post->save();

   return response()->json([
    'message' => 'post created',
    'post description'=>$post,
    'created with'=> User::find(1)->name,
    'email'=> User::find(1)->email,
   ]);

  }

  public function update($id , Request $request){

    $post=Post::find($id);
    $request->validate([
        'title'=>'string',
        'description'=>'required|string',
    ]);
    if($post->user_id == \Auth::user()->id){
    $post->update([
        'title'=>$request->title,
        'description'=>$request->description
    ]);} else{
        return response()->json([
          'message'=>'not allowed to update others post'
        ]);
    }

    return response()->json([
     'status'=>'post updated',
     'updated post'=>$post
    ]);
  }

  public function destroy($id){

    $user_id=\Auth()->user()->id;

    $post=Post::find($id);




    if($post->user_id!= $user_id){
        return response()->json([
            'message'=> 'not allowed to delete others post'
        ]);
    }

    $post->comments()->delete();
    $post->comments()->likes();

    $post->delete();


    $post->delete();
    return response()->json([
        'status'=>true,
        'comment'=>'all comments deleted',
        'Post'=>'post deleted'
      ]);
  }
}

File: mobile/lib/Controllers/PostController.dart


import 'dart:convert';

PostController postFromJson(String str) => PostController.fromJson(json.decode(str));

String postToJson(PostController data) => json.encode(data.toJson());

class PostController {
  Post post;
  int allLikes;
  List<dynamic> likes;

  PostController({
    required this.post,
    required this.allLikes,
    required this.likes,
  });

  factory PostController.fromJson(Map<String, dynamic> json) => PostController(
    post: Post.fromJson(json["post"]),
    allLikes: json["all likes"],
    likes: List<dynamic>.from(json["likes"].map((x) => x)),
  );

  Map<String, dynamic> toJson() => {
    "post": post.toJson(),
    "all likes": allLikes,
    "likes": List<dynamic>.from(likes.map((x) => x)),
  };
}

class Post {
  int id;
  String title;
  String description;
  int userId;
  dynamic image;
  DateTime createdAt;
  DateTime updatedAt;

  Post({
    required this.id,
    required this.title,
    required this.description,
    required this.userId,
    required this.image,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Post.fromJson(Map<String, dynamic> json) => Post(
    id: json["id"],
    title: json["title"],
    description: json["description"],
    userId: json["user_id"],
    image: json["image"],
    createdAt: DateTime.parse(json["created_at"]),
    updatedAt: DateTime.parse(json["updated_at"]),
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "title": title,
    "description": description,
    "user_id": userId,
    "image": image,
    "created_at": createdAt.toIso8601String(),
    "updated_at": updatedAt.toIso8601String(),
  };
}
